src/components/button/index.ts

```diff
@@ -0,0 +1,38 @@
+import Block from '../../utils/block';
+
+import './button.css';
+
+interface ButtonProps {
+    label: string;
+    background: string;
+    styles: string;
+    type: string;
```

Comment:
Для таких данных, как тип чего-то, ```string``` не очень подходит. Тип - это ограниченное количество значений. ```string``` допускает любые значения. Тут лучше подойдет использование union'a
```suggestion
  type?: "button" | "submit";
```